### ISSUE TITLE
Nice error message if the `VirtualView` throws

### ISSUE THREAD

**Issue by kaste** [SublimeLinter/SublimeLinter#1379]
If the linter reports errors referring lines not in the `view`, we get a traceback here

```
Linter crashed.

Traceback (most recent call last):
  File "C:\Users\marce\AppData\Roaming\Sublime Text 3\Packages\SublimeLinter\lint\backend.py", line 87, in execute_lint_task
    errors = linter.lint(code, view_has_changed, settings) or []
  File "C:\Users\marce\AppData\Roaming\Sublime Text 3\Packages\SublimeLinter\lint\linter.py", line 726, in lint
    return self.parse_output(output, virtual_view)
  File "C:\Users\marce\AppData\Roaming\Sublime Text 3\Packages\SublimeLinter\lint\linter.py", line 747, in parse_output
    return self.parse_output_via_regex(output, virtual_view)
  File "C:\Users\marce\AppData\Roaming\Sublime Text 3\Packages\SublimeLinter\lint\linter.py", line 767, in parse_output_via_regex
    error = self.process_match(m, virtual_view)
  File "C:\Users\marce\AppData\Roaming\Sublime Text 3\Packages\SublimeLinter\lint\linter.py", line 848, in process_match
    start, end = vv.full_line(m.line)
  File "C:\Users\marce\AppData\Roaming\Sublime Text 3\Packages\SublimeLinter\lint\linter.py", line 83, in full_line
    start = self._newlines[line]
IndexError: list index out of range
```

Usually, this means the linter reports errors from other files as well. (Or, the regex is broken.)

We should show a nicer error message here.

**Comment by mheinzler** [comment]
I'm having the same problem with clang-tidy. There are a lot of warnings that can appear in header files which sometimes results in this error when the header file has more lines than the linted file.

A workaround for me is to change the mentioned line to the following. At least then the warnings are still shown.
```py
start = self._newlines[min(line, len(self._newlines) - 1)]
```

Of course, ultimately it would be great if SublimeLinter was able to handle warnings/errors from multiple files, but I have no idea of where to start implementing something like this.

**Comment by kaste** [comment]
Shouldn't the linter `clang-tidy` filter messages coming from other files? How can the reported 'line:col' pairs be of any use if applied to the wrong buffer. 

Be forgiving here is usually correct. However, I always thought we need to inform the plugin author that there is something fishy going on here. 

I'll take and would appreciate a PR with your line, but keep in mind that writing the unit tests for such a line is the actual effort.

For multi files support, the first step here is to allow to capture a `filename` via the regex and to pass it up. The biggest effort here is that we cannot just add such a new property to `LintMatch` which is still a tuple bc that would be likely breaking. The newer backend itself accepts new properties bc it is based on simple mappings/dicts.

**Comment by kaste** [comment]
The unit test should be actually easy, because iirc 🙄  I already test for the current behavior. So you just change the line, run the tests, and this will point you to the correct unit test.

**Comment by mheinzler** [comment]
The `clang-tidy` linter could of course filter out these other files. But normally you would want to see these messages, because the `clang-tidy` tool itself can't easily be used on such header files alone.

I can provide a PR in the next days for this. It's not nice but at least for me it is good enough for now to see that there is something wrong.

Maybe I can also take a closer look at real multi files support.
